temporal-sdk/src/main/java/io/temporal/client/WorkflowInvocationHandler.java

```diff
@@ -91,17 +83,18 @@ static <T> void initAsyncInvocation(InvocationType type, T value) {
     } else if (type == InvocationType.EXECUTE) {
       invocationContext.set(new ExecuteWorkflowInvocationHandler());
     } else if (type == InvocationType.SIGNAL_WITH_START) {
-      SignalWithStartBatchRequest batch = (SignalWithStartBatchRequest) value;
+      SignalWithStartBatchRequest batch = (SignalWithStartBatchRequest) values[0];
       invocationContext.set(new SignalWithStartWorkflowInvocationHandler(batch));
     } else if (type == InvocationType.START_NEXUS) {
-      NexusStartWorkflowRequest request = (NexusStartWorkflowRequest) value;
+      NexusStartWorkflowRequest request = (NexusStartWorkflowRequest) values[0];
       invocationContext.set(new StartNexusOperationInvocationHandler(request));
     } else if (type == InvocationType.UPDATE) {
-      UpdateOptions<?> updateOptions = (UpdateOptions) value;
+      UpdateOptions<?> updateOptions = (UpdateOptions<?>) values[0];
       invocationContext.set(new UpdateInvocationHandler(updateOptions));
     } else if (type == InvocationType.UPDATE_WITH_START) {
-      UpdateWithStartWorkflowOperation operation = (UpdateWithStartWorkflowOperation) value;
-      invocationContext.set(new UpdateWithStartInvocationHandler(operation));
+      UpdateOptions<?> options = (UpdateOptions<?>) values[0];
```

Comment:
I would make a module private type to pass these two values over making this whole function take a vararg